theano/sandbox/cuda/var.py

```diff
@@ -211,3 +213,40 @@ def float32_shared_constructor(value, name=None, strict=False,
     rval.get_value_return_ndarray = get_value_return_ndarray
 
     return rval
+
+
+def CudaNdarraySharedVariable_unpickler(*npa):
+    if (config.experimental.unpickle_shared_gpu_on_cpu and
+        config.device == 'cpu'):
+        # directly return numpy array
+        warnings.warn(
+            "config.experimental.unpickle_shared_gpu_on_cpu is set to True."
+            " Unpickling CudaNdarraySharedVariable as TensorSharedVariable."
+        )
+        cls = theano.tensor.sharedvar.TensorSharedVariable
+        type = theano.tensor.TensorType(
+            dtype=npa[1].dtype, broadcastable=npa[1].broadcastable)
+        return cls(name=npa[0],
```

Comment:
If this breaks old pickled files, then they were already broken by the regular code, so I don't think this is a major issue.